cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/PrintIR.qll

```diff
@@ -427,11 +427,18 @@ query predicate edges(PrintableIRBlock pred, PrintableIRBlock succ, string key,
     succBlock = succ.getBlock() and
     predBlock.getSuccessor(kind) = succBlock and
     (
-      (
-        key = "semmle.label" and
+      key = "semmle.label" and
+      exists(string kinds |
+        kinds =
+          concat(EdgeKind k |
```

Comment:
If this is changed to a `strictcount` you could remove line 428, right?

Reply:
In the `key = "semmle.order"` case? That's bound by `getSuccessorIndex` anyway. Although, I do see that you need to rewrite the `exists` slightly to get rid of the resulting cartesian product with `kind`.

Reply:
> In the `key = "semmle.order"` case? That's bound by `getSuccessorIndex` anyway.

Ah, you're right.

So are you thinking of something like:
```ql
query predicate edges(PrintableIRBlock pred, PrintableIRBlock succ, string key, string value) {
  exists(IRBlock predBlock, IRBlock succBlock |
    predBlock = pred.getBlock() and
    succBlock = succ.getBlock() and
    (
      key = "semmle.label" and
      exists(string kinds |
        kinds =
          strictconcat(EdgeKind k |
            predBlock.getSuccessor(k) = succBlock
          |
            k.toString(), "|" order by k.toString()
          )
      |
        if predBlock.getBackEdgeSuccessor(_) = succBlock
        then value = kinds + " (back edge)"
        else value = kinds
      )
      or
      key = "semmle.order" and
      value = getSuccessorIndex(predBlock, succBlock).toString()
    )
  )
}
```

Reply:
Exactly 👍

Reply:
Done in f9bfd969a99cd7b2238e921a09d97088fd4e09eb